cpp/src/IceSSL/SChannelTransceiverI.cpp

```diff
@@ -292,29 +74,58 @@ SChannel::TransceiverI::getNativeInfo()
 IceInternal::SocketOperation
 SChannel::TransceiverI::sslHandshake()
 {
-    DWORD flags = ASC_REQ_SEQUENCE_DETECT | ASC_REQ_REPLAY_DETECT | ASC_REQ_CONFIDENTIALITY | ASC_REQ_ALLOCATE_MEMORY |
-                  ASC_REQ_STREAM;
+    DWORD flags = 0;
     if (_incoming)
     {
-        flags |= ASC_REQ_EXTENDED_ERROR;
-        if (_engine->getVerifyPeer() > 0)
-        {
-            flags |= ASC_REQ_MUTUAL_AUTH;
-        }
+        flags |= ASC_REQ_SEQUENCE_DETECT | ASC_REQ_REPLAY_DETECT | ASC_REQ_CONFIDENTIALITY | ASC_REQ_ALLOCATE_MEMORY |
+                 ASC_REQ_STREAM | ASC_REQ_EXTENDED_ERROR;
     }
     else
     {
-        flags |= ISC_REQ_USE_SUPPLIED_CREDS | ISC_REQ_MANUAL_CRED_VALIDATION | ISC_RET_EXTENDED_ERROR;
+        flags |= ISC_REQ_SEQUENCE_DETECT | ISC_REQ_REPLAY_DETECT | ISC_REQ_CONFIDENTIALITY | ISC_REQ_ALLOCATE_MEMORY |
+                 ISC_REQ_STREAM | ISC_REQ_USE_SUPPLIED_CREDS | ISC_RET_EXTENDED_ERROR;
     }
 
     SECURITY_STATUS err = SEC_E_OK;
     if (_state == StateHandshakeNotStarted)
     {
+        SCHANNEL_CRED cred = _engine->newCredentialsHandle(_incoming);
+        _certificate = _localCertificateSelectionCallback(_incoming ? _adapterName : _host);
+        if (_certificate)
+        {
+            cred.cCreds = 1;
+            cred.paCred = &_certificate;
+        }
+
+        if (_rootStore)
+        {
+            cred.hRootStore = _rootStore;
+        }
+
+        memset(&_credentials, 0, sizeof(_credentials));
+
+        err = AcquireCredentialsHandle(
+            0,
+            const_cast<char*>(UNISP_NAME),
+            (_incoming ? SECPKG_CRED_INBOUND : SECPKG_CRED_OUTBOUND),
+            0,
+            &cred,
+            0,
+            0,
+            &_credentials,
+            0);
```

Comment:
Can any of these be `nullptr`, makes it easier to distinguish numbers from pointers.

Reply:
I would check.